src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -982,6 +983,83 @@ class TcgStatics {
     }
   }
 
+  static holon_pokemon_energy(Object delegate, Integer rainbowEnergyCount) {
+    delegate.globalAbility { Card thisCard->
+      delayed {
+          before PLAY_CARD, {
+            def validPokemon = thisCard.player.pbg.all.findAll {it.cards.energyCount()}
+            def holonAlreadyPlayed = bg.em().retrieveObject("Holon_Pokemon_Energy") == bg.turnCount
+            if (ef.cardToPlay == thisCard) {
+              if (choose([1,2], ["Pokémon", "Energy"], "Play this card as a Pokémon or as an energy?") == 2) {
+                if (!holonAlreadyPlayed) {
+                  if (validPokemon) {
+                    def energyCard
+                    def energyEquivalent = []
+                    def typeImages = []
+                    rainbowEnergyCount.times {
+                      energyEquivalent.add([R, D, F, G, W, L, M, P, Y])
+                      typeImages.add(RAINBOW)
+                    }
+
+                    energyCard = specialEnergy(new CustomCardInfo(thisCard.staticInfo).setCardTypes(ENERGY, SPECIAL_ENERGY), energyEquivalent) {
+                      typeImagesOverride = typeImages
+                      onPlay {}
+                      onRemoveFromPlay {
+                        bg.em().run(new ChangeImplementation(thisCard, energyCard))
+                      }
+                    }
+                    energyCard.player = thisCard.player
+
+                    def pcs = validPokemon.select("Attach to?")
+
+                    // Select an energy before attachment so that thisCard doesn't show up as an option
+                    def returningEnergy = pcs.cards.filterByType(ENERGY).select(count: 1)
```

Comment:
As an aside, filterByType is good for checking if a card exists, but I don't think we should use it in selects. It can be helpful to see all of the cards attached to a PCS when using a select, and the filter of the select can be used to ensure only the correct cards can be selected.

```suggestion
                    def returningEnergy = playEnergy.attached.cards.select cardTypeFilter(ENERGY)
```